js/main.js

```diff
@@ -0,0 +1,153 @@
+//---------------------- Add to cart --------------------------------
+const addEventBtn = (products) => {
+  const btnOrder = document.querySelectorAll('.add-btn');
+  
+  console.log(btnOrder);
+
+  btnOrder.forEach(button => {
+    button.addEventListener('click', () => {
+      const productCategory = button.getAttribute('data-category');
+      const productId = button.getAttribute('data-id');
+
+      const product = products.find(item => item.id === +productId);
+      console.log(product);
+
+      const cartItems = JSON.parse(localStorage.getItem('cart'));
+
+      if(cartItems && cartItems.length > 0) {
+        const existedItem = cartItems.find(item => item.id === product.id && item.title === product.title);
+
+        console.log(existedItem);
+
+        if(existedItem) {
+          const updatedCart = cartItems.map(item => {
+            if(item.id === product.id && item.title === product.title) {
+              console.log(cartItems);
+              return {...item, amount: Number(item.amount) + 1}
+            }
+            return item;
+          });
+          console.log(updatedCart);
+          localStorage.setItem('cart', JSON.stringify(updatedCart));
+          const quantity = updatedCart.reduce((total, item) => total + item.amount, 0);
+          console.log(quantity);
+          localStorage.setItem('quantity', JSON.stringify(quantity));
+          document.querySelector('#cart-number').innerHTML = quantity;
```

Comment:
1 hàm của anh nó dài quá đi, nếu nó đã dài hơn 15-20 dòng là đã có vấn đề rồi.
Đoạn `btnOrder.forEach(button => { button.addEventListener('click', () => {...})`
Thì anh có thể tách ra thành 1 hàm riêng, hàm handleAddToCart chẳng hạn
`btnOrder.forEach(button => { button.addEventListener('click', handleAddToCart)`
Cái thứ 2 là đoạn code này với đoạn else ở dưới y hệt nhau, đang bị lặp code. Và cái này là theo như em thấy thì nó là tổng giá tiền, thì anh tách ra 1 hàm riêng để xử lý, lấy products trên localStorage rồi gán trực tiếp vào luôn, k cần phải `localStorage.setItem('quantity')` làm gì, vì điều này k có ý nghĩa.
Dù gì thì khi cập nhật 1 sản phẩm ở giỏ hàng, thì nó cũng sẽ diễn ra trong các bước add to cart => update amount => update total price => set localStorage => set innerHTML bằng cái total price

Thì hoàn toàn có thể như này mà
add to cart => update amount => update total price => set innerHTML